docs/user/app_feature_config_to_push.md

```diff
@@ -0,0 +1,99 @@
+# Configuration Postprocessing
+
+!!! note
+    current implementation **only renders the configuration to push, it doesn't update the configuration** into the target devices.
+
+The intended configuration job doesn't produce a final configuration artifact(see below for reasons why). Actually, you should understand it as the "intended" **running** configuration, because the intended configuration job generates what is in the final running configuration. This works well for the "compliance" feature, but not to create a configuration artifact that is ready to push.
+
+Limitations of the intended configuration feature:
+
+- Because the intended configuration is stored in the database, and in an external Git repository, it SHOULD NOT contain any secret (or derivative).
+- The format of the running configuration is not always the same as the configuration to push, examples include:
+  - Pushing snmpv3 configurations
+  - VTP configurations
+  - Implicit configurations like a "no shutdown" on an interface
+
+As the Golden Config application becomes more mature in delivering an all encompassing configuration management solution, it requires an advanced feature to render a configuration artifact. That artifact must be in the final format your device is expecting, from the intended configuration.
+
+This is exposed via the `get_config_postprocessing()` function exposed in `nautobot_golden_config.utilities.config_postprocessing`. This method takes the current configurations generated by the Golden Config intended configuration feature, and the HTTP request. This function will return the intended configuration that is **ready to push**.
+
+From the user perspective, you can retrieve this configuration via two methods:
+
+- UI: within the `Device` detail view, if the feature is enabled, a new row in the "Configuration Types" appears, and clicking the icon the new configuration will be rendered on the fly (synchronously). Check figure.
+- REST API: at the path `/api/plugins/golden-config/config-postprocessing/{device_id}` you can request the intended configuration processed, and the return payload will contain a "config" key with the rendered configuration.
+
+![Configuration Postprocessing](../images/config_postprocessing_1.png)
+
+## Customize Configuration Processing
+
+There are two different ways to customize the default behavior of `get_config_postprocessing` method:
+
+- `config_postprocessing_callable`: is the list of **available methods** for processing the intended configuration. It contains some default implemented methods, currently `render_secrets`. But it could be extended via configuration options (see next section).
```

Comment:
We need to provide context that it should be in the dotted string format to an importable path.

Reply:
ping.

Reply:
resolved in 5b88c8afa091542b049f6de527ecb0d4001df50b